auditbeat/magefile.go

```diff
@@ -0,0 +1,212 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/dev-tools/mage"
+)
+
+func init() {
+	mage.BeatDescription = "Audit the activities of users and processes on your system."
+}
+
+// Build builds the Beat binary.
+func Build() error {
+	return mage.Build(mage.DefaultBuildArgs())
+}
+
+// GolangCrossBuild build the Beat binary inside of the golang-builder.
+// Do not use directly, use crossBuild instead.
+func GolangCrossBuild() error {
+	return mage.GolangCrossBuild(mage.DefaultGolangCrossBuildArgs())
+}
+
+// BuildGoDaemon builds the go-daemon binary (use crossBuildGoDaemon).
+func BuildGoDaemon() error {
+	return mage.BuildGoDaemon()
+}
+
+// CrossBuild cross-builds the beat for all target platforms.
+func CrossBuild() error {
+	return mage.CrossBuild()
+}
+
+// CrossBuildGoDaemon cross-builds the go-daemon binary using Docker.
+func CrossBuildGoDaemon() error {
+	return mage.CrossBuildGoDaemon()
+}
+
+// Clean cleans all generated files and build artifacts.
+func Clean() error {
+	return mage.Clean()
+}
+
+// Package packages the Beat for distribution.
+// Use SNAPSHOT=true to build snapshots.
+// Use PLATFORMS to control the target platforms.
+func Package() {
+	start := time.Now()
+	defer func() { fmt.Println("package ran for", time.Since(start)) }()
+
+	mage.UseElasticBeatPackaging()
+	customizePackaging()
+
+	mg.Deps(Update)
+	mg.Deps(makeConfigTemplates, CrossBuild, CrossBuildGoDaemon)
+	mg.SerialDeps(mage.Package, TestPackages)
+}
+
+// TestPackages tests the generated packages (i.e. file modes, owners, groups).
+func TestPackages() error {
+	return mage.TestPackages()
+}
+
+// Update updates the generated files (aka make update).
+func Update() error {
+	return sh.Run("make", "update")
+}
+
+// -----------------------------------------------------------------------------
+// Customizations specific to Auditbeat.
+// - Config files are Go templates.
+
+const (
+	configTemplateGlob      = "module/*/_meta/config*.yml.tmpl"
+	shortConfigTemplate     = "build/auditbeat.yml.tmpl"
+	referenceConfigTemplate = "build/auditbeat.reference.yml.tmpl"
+)
+
+func makeConfigTemplates() error {
+	configFiles, err := mage.FindFiles(configTemplateGlob)
+	if err != nil {
+		return errors.Wrap(err, "failed to find config templates")
+	}
+
+	var shortIn []string
+	shortIn = append(shortIn, "_meta/common.p1.yml")
+	shortIn = append(shortIn, configFiles...)
+	shortIn = append(shortIn, "_meta/common.p2.yml")
+	shortIn = append(shortIn, "../libbeat/_meta/config.yml")
+	if !mage.IsUpToDate(shortConfigTemplate, shortIn...) {
+		fmt.Println(">> Building", shortConfigTemplate)
```

Comment:
Why didn't you extracted this 4 lines into a separate function with parameters `template`, `snippets`, `permission` which passes the correct permissions to `MustFileConcat`? I think this would look cleaner:

```golang
if !mage.IsUpToDate(shortConfigTemplate, shortIn...) {
    updateConfigTemplate(shortConfigTemplate, shortIn, 0600)
}

if !mage.IsUpToDate(referenceConfigTemplate, referenceIn...) {
   updateConfigTemplate(referenceConfigTemplate, referenceIn, 0644)
}

// ...
```
The new extracted function could be:
```golang
func updateConfigTemplates(template string, snippets []string, permission int) {
    fmt.Println(">> Building", template)
    mage.MustFileConcat(template, 0644, snippets...)
    mage.MustFindReplace(template, regexp.MustCompile("beatname"), "{{.BeatName}}")
    mage.MustFindReplace(template, regexp.MustCompile("beat-index-prefix"), "{{.BeatIndexPrefix}}")
}
```

So if the way config files are updated is changed, we only need to modify it in one function.